exercises/practice/all-your-base/.approaches/introduction.md

```diff
@@ -0,0 +1,68 @@
+# Introduction
+
+The process of the _all your base_ exercise can be divided into  up to four parts:
+- conversion from input digit vector into a base 10 number
+- conversion from intermediate number into target base vector
+- possible reversing of the resulting vector
+- argument error checking
```

Comment:
There's also the interesting approach of doing the conversion directly, without an intermediate numeric value.
You need two functions that are usually found in "arbitrary-precision integer" a.k.a. "BigInteger" implementations: 
- a function `multiply()` that takes a `std::vector` of digits (in `out_base`) and the `in_base`, and multiplies the digits with the `in_base`.
- a function `add()` that takes a `std::vector` of digits (in `out_base`) and a single digit in `in_base` and adds the value of this single digit to the digits.